app/settings.py

```diff
@@ -1,4 +1,11 @@
+import json
 import os
+from typing import List
 
-API_TOKEN = os.getenv("API_TOKEN")
-USER_ID = int(user_id) if (user_id := os.getenv("USER_ID")) else None
+def _parse_env(key: str, default):
+    return json.loads(value) if (value := os.getenv(key)) else default
+
+API_TOKEN: str = os.getenv("API_TOKEN")
+ALLOWED_IDS: List[int] = _parse_env('ALLOWED_IDS', [])
```

Comment:
Python 3.9 (and up) supports direct list annotations, which not requiring any imports from typing
```python
ALLOWED_IDS: list[int] = _parse_env('ALLOWED_IDS', [])
```

Though, I probably want to refactor this section to a class like this
```python
@dataclass
class Settings:
    api_token: str
    allowed_ids: list[int]
    reply_to_message: bool
    with_captions: bool
```

And to fully support typing, it would be better to have few parse functions
```python
def parse_env_list(key: str) -> list[int]:
    return json.loads(os.getenv(key, "[]"))

def parse_env_bool(key: str) -> bool:
    return os.getenv(key, "false").lower() in ("yes", "true", "1", "on")
 ```
 
To have something like this
 ```python
 settings = Settings(
     api_token = os.getenv("API_TOKEN", ""),
     allowed_ids = parse_env_list("ALLOWED_IDS"),
     reply_to_message = parse_env_bool("REPLY_WITH_MESSAG"),
     with_captions = parse_env_bool("WITH_CAPTIONS")
 )
 ```

Reply:
Fixed in the latest commit